source/time_tables/tests/departure_boards_test.cpp

```diff
@@ -2390,3 +2390,352 @@ BOOST_AUTO_TEST_CASE(schedules_on_circular_routes) {
         BOOST_CHECK_EQUAL(teminus_schedule.date_times(1).time(), time_to_int(10, 15, 00));
     }
 }
+
+BOOST_AUTO_TEST_CASE(terminus_schedule_group_by_stoparea) {
+    /*
+     * 1 line, 1 routes and 2 VJs
+     * A: StopArea with StopPoints Avj1 and Avj2
+     * VJ1 : D->C->B->AVj1
+     * VJ2 : C->B->AVj2
+     *            ----------------
+     *           |   Avj2 <----   |
+     * Route1    |             |  |
+     *           |   Avj1 <-------------- B <- C <- D
+     *           |                |
+     *            ----------------
+     * Calculate terminus schedule at StopArea (C)
+     *
+     */
+    ed::builder b("20160101");
+    b.sa("A", 0., 0., false)("Avj1")("Avj2");
+    b.sa("B", 0., 0., false)("B:sp");
+    b.sa("C", 0., 0., false)("C:sp");
+    b.sa("D", 0., 0., false)("D:sp");
+
+    b.vj("Line1").route("Route1").name("VJ1")("D:sp", "08:00"_t)("C:sp", "09:00"_t)("B:sp", "10:00"_t)("Avj1",
+                                                                                                       "11:00"_t);
+    b.vj("Line1").route("Route1").name("VJ2")("C:sp", "09:10"_t)("B:sp", "10:10"_t)("Avj2", "11:10"_t);
+
+    b.finish();
+    b.data->pt_data->sort_and_index();
+    b.data->build_raptor();
+    b.data->pt_data->build_uri();
```

Comment:
```suggestion
    b.make();
```
`b.make()` does all the heavy lifting for you.
Copy/pasting section of code should be a red flag for you !

Reply:
ok